--- a/cmake/decomp2d/downloadBuild2decomp.cmake.in
+++ b/cmake/decomp2d/downloadBuild2decomp.cmake.in
@@ -11,6 +11,8 @@ ExternalProject_Add(downloadBuild2decomp
     GIT_TAG           "v2.0.4"
     SOURCE_DIR        "${CMAKE_CURRENT_BINARY_DIR}/decomp2d-src"
     INSTALL_DIR       "${CMAKE_CURRENT_BINARY_DIR}/decomp2d-opt"
+    PATCH_COMMAND     ${CMAKE_COMMAND} -E chdir ${CMAKE_CURRENT_BINARY_DIR}/decomp2d-src
+                      patch -p1 < /mnt/common/hxu40/coeus/derived/coeus-adapter/CI/coeus/packages/incompact3D/patches/decomp_fix.patch
     TEST_COMMAND      ""
 )
