diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8f2b359..c7f3c82 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,10 +1,10 @@
 cmake_minimum_required(VERSION 3.20)                        
 cmake_policy(SET CMP0074 NEW)
 
-project(decomp2d
-  LANGUAGES Fortran)
+project(decomp2d LANGUAGES C CXX Fortran)
 set(version 2.0.4)
 enable_testing()
+find_package(adios2 REQUIRED)
 if (IO_BACKEND MATCHES "adios2")
   # Can be useful to also activate CXX, sometimes is needed by packages
   enable_language(C CXX)
@@ -34,7 +34,7 @@ option(ENABLE_INPLACE "Enable in-place operation of FFT" OFF)
 option(HALO_DEBUG "Enable halo debugging options" OFF)
 option(EVEN "Padded alltoall transpose operations" OFF)
 option(BUILD_SHARED_LIBS "Set ON to build a shared library" OFF)
-set (IO_BACKEND "mpi" CACHE STRING "Default IO backend (mpi (default) or adios2)")
+set(IO_BACKEND "adios2" CACHE STRING "Default IO backend (mpi or adios2)" FORCE)
 set_property(CACHE IO_BACKEND PROPERTY STRINGS mpi adios2)
 
 set(AUTHOR "Stefano Rolfo;Charles Moulinec;Paul Bartholomew")
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 2e80684..6c84d17 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -5,7 +5,7 @@ file(GLOB files_decomp factor.f90
 		       log.f90
 		       io.f90)
 include_directories(${CMAKE_SOURCE_DIR}/src)
-
+find_package(adios2 REQUIRED)
 if(${BUILD_TARGET} MATCHES "gpu")
   list(APPEND files_decomp decomp_2d_cumpi.f90)
   if(ENABLE_NCCL)
@@ -79,10 +79,10 @@ if (HALO_DEBUG)
   target_compile_definitions(decomp2d PUBLIC -DHALO_DEBUG)
 endif()
 
-if (IO_BACKEND MATCHES "adios2")
-  target_compile_definitions(decomp2d PUBLIC -DADIOS2)
-  target_link_libraries(decomp2d PUBLIC adios2::fortran_mpi adios2::fortran)
-endif (IO_BACKEND MATCHES "adios2")
+#if (IO_BACKEND MATCHES "adios2")
+target_compile_definitions(decomp2d PUBLIC -DADIOS2)
+target_link_libraries(decomp2d PUBLIC adios2::fortran_mpi adios2::fortran)
+  #endif (IO_BACKEND MATCHES "adios2")
 
 install(TARGETS decomp2d
   EXPORT decomp2d-targets
